tools/snap/build_remote.py

```diff
@@ -266,11 +263,16 @@ def main():
             if process.is_alive():
                 raise ValueError(f"Timeout out reached ({args.timeout} seconds) during the build!")
 
-            workspaces = {}
+            build_success = True
             for async_result in async_results:
-                workspaces.update(async_result.get())
-
-            if _dump_results(targets, archs, status, workspaces):
+                if not async_results.get():
+                    build_success = False
+
+            _dump_results(archs, status)
```

Comment:
With this refactoring, `_dump_failed_build_logs` is effectively a noop if you invoke it when there is no failures. I think we can simplify the logs handling a little further: what about calling inconditionally `_dump_failed_build_logs` just before `_dump_results`, and remove its call (and the necessity to set `dump_output` variable) in the `_build_snap` function?

Reply:
To do that, I think I'd have to go back to returning dicts from `_build_snap`. If we stopped calling it in `_build_snap`, we'd only print the last failed build log instead of printing all of them which I personally prefer we do to avoid throwing away error information. Finally, I think we'd still need `dump_output` in `_build_snap` if we wanted to continue dumping `snapcraft`'s output on failures which we did before this PR and I think is nice.

Reply:
Ah yes, I was thinking of a quick win by iterate over the targets, but there is still the workspace to infer. In this case it would not simplify things significantly.